### DISCUSSION TITLE
v0.6.0: What will change?

### DISCUSSION THREAD

**Discussion by GogoVega** [GogoVega/node-red-contrib-firebase-realtime-database#50] [Announcements]
# 0.6.0: Breaking Changes

**TL;DR**: This new version introduces a breaking change by externalizing the config-node which involves user action.

## Why?

My ambition is to support database management, [Cloud Firestore](https://firebase.google.com/docs/firestore), [Cloud Storage](https://firebase.google.com/docs/storage) and [Real-Time](https://firebase.google.com/docs/database). The real-time database is already operational because it's this module. The Cloud Firestore database is under development. I have not yet started work on the Cloud Storage database.

To achieve this goal, I need to create a shared config-node between these three databases which involves outsourcing this config-node. It serves as a client and connects to Firebase. This config-node is available [here](https://github.com/GogoVega/Firebase-Config-Node).

Which brings us to our breaking change:

## Breaking Changes

The config-node type changes from `database-config` to `firebase-config` which results in a `Type missing` error (see screenshot below) in the editor if this change has not been made in your flows file.

<img width="491" alt="Capture d’écran 2023-10-18 à 13 35 53" src="https://github.com/GogoVega/node-red-contrib-firebase-realtime-database/assets/92022724/bd1acc60-4e9e-4f91-bf5c-c030aad35ef8">

To do this, a Migration Wizard has been added. Please read more [here](https://github.com/GogoVega/node-red-contrib-firebase-realtime-database/wiki/migration-wizard#how-to-migrate).

---
A full list of changes can be read [here](https://github.com/GogoVega/node-red-contrib-firebase-realtime-database/milestone/7).

If you have any questions, comments or problems, please comment 🙂

**Comment by GogoVega** [comment]
Final Beta version as [0.6.0-beta.5](https://github.com/GogoVega/node-red-contrib-firebase-realtime-database/releases/tag/v0.6.0-beta.5) has just been released 🔥 

Still to do before the production release:
- check the behavior of the `RED.history` with config nodes (Migration)
- check/update all the documentation

**Reply by GogoVega** [reply]
[v0.6.0 Released](https://github.com/GogoVega/node-red-contrib-firebase-realtime-database/releases/tag/v0.6.0) 🔥
